panther_analysis_tool/main.py

```diff
@@ -404,8 +404,69 @@ def upload_analysis(args: argparse.Namespace) -> Tuple[int, str]:
     return 0, ""
 
 
+def get_analysis_id_by_query(args: argparse.Namespace, query_list: list) -> list:
+    query_list_payload = {"listDetections": {"scheduledQueries": query_list}}
+    client = get_client(args.aws_profile, "lambda")
+
+    query_info = client.invoke(
+        FunctionName="panther-analysis-api",
+        InvocationType="RequestResponse",
+        LogType="None",
+        Payload=json.dumps(query_list_payload),
+    )
+
+    if query_info["ResponseMetadata"]["HTTPStatusCode"] != 200:
+        logging.warning(
+            "Failed to search for associated queries, API error.\n\t status code: %s",
+            query_info["ResponseMetadata"]["HTTPStatusCode"],
+        )
+        return []
+
+    analysis_api_response = json.loads(query_info["Payload"].read().decode("utf-8"))
+    analysis_json_response = analysis_api_response["body"]
+
+    analysis_json_output = json.loads(analysis_json_response)
+
+    analysis_id_list = []
+    for detection in analysis_json_output.get("detections"):
+        analysis_id_list.append(detection.get("id"))
+
+    return analysis_id_list
+
+
+def get_query_by_analysis_id(args: argparse.Namespace, analysis_id_list: list) -> list:
+    analysis_list_payload = {"listDetections": {"ids": analysis_id_list}}
+    client = get_client(args.aws_profile, "lambda")
+
+    analysis_info = client.invoke(
+        FunctionName="panther-analysis-api",
+        InvocationType="RequestResponse",
+        LogType="None",
+        Payload=json.dumps(analysis_list_payload),
+    )
+
+    if analysis_info["ResponseMetadata"]["HTTPStatusCode"] != 200:
+        logging.warning(
+            "Failed to search for associated queries, API error.\n\t status code: %s",
+            analysis_info["ResponseMetadata"]["HTTPStatusCode"],
+        )
+        return []
+
+    analysis_api_response = json.loads(analysis_info["Payload"].read().decode("utf-8"))
+    analysis_json_response = analysis_api_response["body"]
+
+    analysis_json_output = json.loads(analysis_json_response)
```

Comment:
I wonder if this could be extracted into a method that each `get_query_by_analysis_id` and `get_analysis_id_by_query` could call?